src/comms/config_layer.cpp

```diff
@@ -184,3 +240,173 @@ void Config::fill_data(CommsPacket packets[MAX_CONFIG_PACKETS], uint8_t sizes[MA
         }
     }
 }
+
+bool ConfigLayer::sd_load() {
+    // num of total bytes in config_packets
+    int config_byte_size = MAX_CONFIG_PACKETS * sizeof(CommsPacket);
+
+    if (sdcard.open("/config.pack") != 0) return false;
+
+    // grab ID from config (originally from hive). should match ID from ref system- if not, abort!!
+    float received_id;
+    sdcard.read((uint8_t*)(&received_id), sizeof(float));
+
+    // checksum is passed by reference and written to for later reference
+    uint64_t checksum;
+    if (!config_SD_read_packets(checksum)) return false;
+
+    sdcard.close();
+
+    uint8_t* config_bytes = (uint8_t*)config_packets;
+#ifdef CONFIG_LAYER_DEBUG
+    Serial.printf("sd_load: computing checksum for stored config\n");
+#endif
+    // need combined checksum of config_bytes, subsec_sizes
+    if ((sd_checksum64(config_bytes, config_byte_size) + sd_checksum64(subsec_sizes, MAX_CONFIG_PACKETS)) != checksum) {     
+        Serial.printf("Checksum for config file does not match stored config, requesting config from hive...\n");
+        return false;
+    }
+
+    Config temp_config;
```

Comment:
Doccument your reason for the temp_config. You explained it to me, but explain it to everyone else reading.